virttest/utils_test/libvirt.py

```diff
@@ -1076,23 +1076,38 @@ def __init__(self):
         # The time spent when migrating vms
         # format: vm_name -> time(seconds)
         self.mig_time = {}
+        # The CmdResult returned from virsh migrate command
+        self.ret = None
 
-    def thread_func_migration(self, vm, desturi, options=None):
+    def thread_func_migration(self, vm, desturi, options=None,
+                              ignore_status=False):
         """
         Thread for virsh migrate command.
 
         :param vm: A libvirt vm instance(local or remote).
-        :param desturi: remote host uri.
+        :param desturi: Remote host uri.
+        :param options: The options for migration command.
+        :param ignore_status: True, means no CmdError will be caught
+                              for the failure.
+                              False, means an CmdError will be caught
+                              for the failure.
         """
         # Migrate the domain.
         try:
             if options is None:
                 options = "--live --timeout=60"
             stime = int(time.time())
-            vm.migrate(desturi, option=options, ignore_status=False,
-                       debug=True)
+            self.ret = vm.migrate(desturi, option=options,
+                                  ignore_status=ignore_status,
+                                  debug=True)
             etime = int(time.time())
             self.mig_time[vm.name] = etime - stime
+            if self.ret.exit_status is not None:
+                logging.debug("Migration to %s returns failed exit status",
+                              desturi)
+                self.RET_LOCK.acquire()
+                self.RET_MIGRATION = False
+                self.RET_LOCK.release()
         except process.CmdError, detail:
             logging.error("Migration to %s failed:\n%s", desturi, detail)
             self.RET_LOCK.acquire()
```

Comment:
The except here just log the error and then release the lock, so why not use ```try...finally``` block instead